src/backends/torch/torchinputconns.cc

```diff
@@ -369,6 +370,232 @@ namespace dd
     return new_dataset;
   }
 
+  // ===== ImgTorchInputFileConn
+
+  void ImgTorchInputFileConn::read_image_folder(
```

Comment:
Is this the same function we use when feeding the Caffe backend ? If yes, shall we move it up to the top image connector class ?

Reply:
There is already a function in DDImg that reads the list of files, but we can't use it yet because it also loads all the image data (and we do not want that if we are using a db)
As modifying DDImg can have unexpected consequences, it would be better to do it in another PR, what do you think?

Reply:
OK. Loading dataset into db from list of files was a user request, it's useful in some settings. It doesn't load in memory though, unless it's using the ImageInputLayer that is internal to caffe, FTR.

Reply:
> > * Is using a test dataset in unit tests a good idea? It could check that the model is actually learning, and that the test process works, but it introduces a small part of random in unit tests, which may not be a good idea?
> 
> Yes, if I understand correctly, you can look at `tests/ut-caffeapi.cc` for how we handle training tests and validate test based on a lower bound on accuracy.

The test checks that the accuracy is greater than 0.6. This lower bound is low enough for the test not to fail randomly, but greater than 0.5 so that if the model output pure random, we can detect it. I think it's a good compromise.
> 
> > *   Test split is not supported, should I include it in this PR?
> 
> Yes it should, usually it's a simple shuffle + split of the list of images in the dataset.

Done